### PR TITLE
GEODE-4968: Further templatizing CMake product name

### PR DESCRIPTION
**Pull request by mcmellawatt**

### CODE REVIEW

File: examples/cmake/FindGeodeNative.cmake.in [hunk 1]
```diff
@@ -114,12 +113,12 @@ if (GeodeNative_FIND_COMPONENTS)
...
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(GeodeNative
+FIND_PACKAGE_HANDLE_STANDARD_ARGS(@PRODUCT_NAME_NOSPACE@
                                   REQUIRED_VARS ${_GEODE_NATIVE_REQUIRED_VARS}
                                   VERSION_VAR GEODE_NATIVE_VERSION_STRING)
 
 # Copy the results to the output variables and target.
-if(GeodeNative_FOUND)
+if(@PRODUCT_NAME_NOSPACE@_FOUND)
   if(NOT TARGET ${GEODE_NATIVE_CPP_TARGET})
     set(GEODE_NATIVE_CPP_TARGET "GeodeNative::cpp")
```
Comment (pivotal-jbarrett) [inline examples/cmake/FindGeodeNative.cmake.in]: Would we not want to parameterize this target? It should match all the other variable names and product.

File: examples/dotnet/PdxAutoSerializer/Program.cs [hunk 2]
```diff
@@ -28,6 +28,10 @@ public static void Main(string[] args)
                 .Set("log-level", "none");
             var cache = cacheFactory.Create();
 
+            Console.WriteLine("Registering for reflection-based auto serialization");
```
Comment (pivotal-jbarrett) [inline examples/dotnet/PdxAutoSerializer/Program.cs]: Why is the example changing in this PR?

Comment (moleske) [reply examples/dotnet/PdxAutoSerializer/Program.cs]: It wasn't fixed in another commit, so the examples were failing to build when testing this

**Comment by mcmellawatt** [comment]
I templatized all the remaining product name/product name derivative variables.  Some of these are internal to the cmake file so it wasn't technically necessary, but I thought being consistent was the best bet in case someone every wants to read the contents of the Find<ProductName> module.

**Pull request merged**